system/scripts/zone/core/calc_combat.cs

```diff
@@ -89,6 +89,23 @@ public float SCR_CalculateDamage(ICombatEntity attacker, ICombatEntity target, S
 
 		var damage = SCR_GetRandomAtk(attacker, target, skill, modifier, skillHitResult);
 
+		// Add extra attack damage based on concentrate
+		// TODO move this to a buff handler
+		if (attacker.Components.Get<BuffComponent>().TryGet(BuffId.Concentrate_Buff, out var concentrateBuff))
```

Comment:
Since you already mentioned moving this to a buff handler, what do you think about putting it in a static method on the buff handlers, similar to what I did with Link? That way the buff-relevant code is all grouped together and this function is easier to read and maintain while we don't have a hook system yet.

Reply:
I can understand that, I've been going back and forth on it as well. Though you could also do the check here.
```cs
if (attacker.IsBuffActive(BuffId.Concentrate_Buff))
    Concentrate.AddBonus(modifier);
```
Or you could name the method something like `TryAddBonus`, indicating that it's not guaranteed, but that it will happen if applicable.

Reply:
I guess it's fine for now, once we have a hooking system it won't matter, since only the methods that relevant will be hooked.